Define Hybrid SRE backplane role for HCP team

mmazur opened the pull request:
### What type of PR is this?
feature

### What this PR does / why we need it?
Give the HCP team access to HCP components on prod clusters.

### Which Jira/Github issue(s) this PR fixes?
Implements [HOSTEDCP-891](https://issues.redhat.com/browse/HOSTEDCP-891)
Depends on https://gitlab.cee.redhat.com/service/backplane-api/-/merge_requests/273  


### Special notes for your reviewer:

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [x] Included documentation changes with PR
- [x] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

cblecker commented on deploy/backplane/hybridsre-hcp/hypershift/management-cluster/20-hybridsre-hcp.SubjectPermission.yml:
```diff
@@ -0,0 +1,13 @@
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - clusterRoleName: hypershift-readers
```
does `dedicated-readers` function as a superset of both these groups?

mmazur replied:
Indeed it does. I should just switch the whole thing to dedicated-readers then?

mmazur replied:
Switched to dedicated-readers.

jewzaam commented:
/retest

jewzaam reviewed (changes requested):
@mmazur new PR check on backplane configs has merged and runs as a default `make` target.  Please rebase, run `make` and fix any issues that are reported.

mmazur commented:
Updated with:
* switched from separate `view` and `hypershift-readers` roles to using `dedicated-readers`, which is a superset of both, per cblecker's suggestion
* rebased and added `namespacesDeniedRegex:` per automated test requirements

jewzaam commented:
/approve

cblecker commented:
/lgtm
/approve
/label tide/merge-method-squash
